src/validation.cpp

```diff
@@ -2795,6 +2795,11 @@ bool CChainState::InvalidateBlock(CValidationState& state, const CChainParams& c
     while (true) {
         if (ShutdownRequested()) break;
 
+        // Make sure the validation callbacks don't overflow
```

Comment:
Commit 1b9518cc7c48750559a39fad4a5056d6a029ee8a

Care to elaborate? Looks like this code could be in a function or something?

Reply:
I've created a small function, and improved the comments ("overflow" wasn't really correct and certainly confusing).